In [12]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [5]:
import mysql.connector as sql

db = sql.connect(
    host="127.0.0.1",
    user="root",
    passwd="root",
    port=3306,
    use_pure=True,
    database="orders",
    auth_plugin="mysql_native_password",
)

print(db)

In [6]:
mycursor = db.cursor()

### Task 1:

1. Write a query to display customer full name with their title (Mr/Ms), both first name and last name are in upper case, customer email id, customer creation date, and display customer's category after applying the below categorization rules:

    i) If customer creation date Year < 2005 Then Category A
    ii) If customer creation date Year >= 2005 and < 2011 Then Category B
    iii) If customer creation date Year >= 2011 Then Category C

    Hint: Use CASE statement, no permanent change in the table required.

    [NOTE: TABLES to be used - ONLINE_CUSTOMER TABLE]


In [14]:
query = """
    SELECT 
        CASE OC.CUSTOMER_GENDER
            WHEN 'M' THEN CONCAT('Mr. ', OC.CUSTOMER_FNAME, ' ', OC.CUSTOMER_LNAME)
            WHEN 'F' THEN CONCAT('Ms. ', OC.CUSTOMER_FNAME, ' ', OC.CUSTOMER_LNAME)
        END AS Full_Name,
        OC.CUSTOMER_EMAIL,
        OC.CUSTOMER_CREATION_DATE,
        CASE
            WHEN OC.CUSTOMER_CREATION_DATE < '2005-01-01' THEN 'Category A'
            WHEN OC.CUSTOMER_CREATION_DATE BETWEEN '2005-01-01' AND '2011-01-01' THEN 'Category B'
            ELSE 'Category C'
        END AS Customer_category  
    FROM orders.online_customer OC
"""

mycursor.execute(query)

In [15]:
results = mycursor.fetchall()

In [17]:
df = pd.DataFrame(
    results,
    columns=[
        "Full_Name",
        "CUSTOMER_EMAIL",
        "CUSTOMER_CREATION_DATE",
        "Customer_category",
    ],
)
df

Full_Name              CUSTOMER_EMAIL  \
0          Ms. Jennifer Wilson             jen_w@gmail.com   
1            Mr. Jackson Davis         dave_jack@gmail.com   
2          Ms. Komal Choudhary        ch_komal@yahoo.co.IN   
3             Mr. Wilfred Jean            w_jean@gmail.com   
4            Ms. Anita Goswami          agoswami@gmail.com   
5           Ms. Ashwathi Bhatt        ash_bhat@yahoo.co.IN   
6         Ms. Neetha Castelina          neetha20@gmail.com   
7            Ms. Devika Satish         devika_sa@gmail.com   
8             Ms. Bidhan C.Roy       bidhanroy@yahoo.co.in   
9                Mr. Vikas Jha         vikas.jha@gmail.com   
10            Mr. Arul Kumar.T         arulkumar@gmail.com   
11          Mr. Ravi Srinivasn     r_srinivasn@yahoo.co.in   
12           Mr. Avinash Dutta   avinash.dutta@yahoo.co.in   
13             Ms. Jyoti Sinha        jyotisinha@gmail.com   
14         Mr. Vijay Bollineni        vbollineni@gmail.com   
15           Mr. Prasad Shetty         pshetty@yahoo.co.in   
16             Mr. Suresh Babu           sbabu@yahoo.co.in   
17          Ms. Bharti Subhash           bhartis@gmail.com   
18              Mr. Keshav Jog          kesjog@yahoo.co.in   
19          Ms. Ramya Ravinder         ramya_r23@gmail.com   
20          Mr. Alan Silvestri         alan_silver@msn.com   
21          Mr. Andrew Stanton    andrew_stanton@yahoo.com   
22            Ms. Anna Pinnock      anna_pinnock@yahoo.com   
23            Mr. Brian Grazer      brian_grazer@gmail.com   
24         Mr. Bruno Delbonnel     bruno_delbonnel@msn.com   
25       Mr. Stephen E. Rivkin   stephen_e. rivkin@msn.com   
26               Ms. Mali Finn         mali_finn@yahoo.com   
27            Mr. Sayyed Faraj      sayyed_faraj@gmail.com   
28        Ms. Francine Maisler  francine_maisler@gmail.com   
29             Ms. Anita Kohli       anita_kohli@yahoo.com   
30           Mr. Thomas Newman       thomas_nman@yahoo.com   
31             Mr. Hans Zimmer       hans_zimmer@yahoo.com   
32          Ms. Niseema Zimmer          niseemaz@yahoo.com   
33             Mr. Hans Zimmer       hans_zimmer@gmail.com   
34           Mr. Thomas Newman     thomas_newman@gmail.com   
35      Ms. Michelle H. Shores     howard_shores@yahoo.com   
36     Mr. James Newton Howard     james_nhoward@yahoo.com   
37           Mr. John Lasseter         john_lass@gmail.com   
38             Ms. Liz Mullane       liz_mullane@gmail.com   
39             Mr. Paul Haggis       paul_haggis@gmail.com   
40  Mr. Tharman Shanmugaratnam        tharshan@yahoo.co.sg   
41             Ms. Rebecca Lim            reblim@msn.co.sg   
42    Mr. Rajiv Chandrasekaran      rajiv_chan@yahoo.co.in   
43              Ms. Tanya Chua           tanyac@singers.sg   
44             Ms. Janvi Rajiv         janvi_jha@msn.co.sg   
45             Ms. Tan Bee Soo       tanbeesoo@yahoo.co.sg   
46                 Mr. Yun Zhu            yunzho@gmail.com   
47  Mr. Wajdi bin Abdul Majeed        wajdiabdul@gmail.com   
48    Ms. Anbara binti Mubarak       anmubarak@yahoo.co.my   
49       Ms. Sri binti  Yaakob        sribinti@yahoo.co.my   
50      Mr. Ahmad Bin Gh Azali     ahmad_bingh@yahoo.co.my   
51   Ms. Suchirithaa Ekanayake          suchiritha@msn.com   

   CUSTOMER_CREATION_DATE Customer_category  
0              1991-06-01        Category A  
1              2001-06-12        Category A  
2              2002-06-26        Category A  
3              2006-01-12        Category B  
4              2006-03-13        Category B  
5              2007-04-15        Category B  
6              2011-08-16        Category C  
7              2011-09-01        Category C  
8              2011-10-23        Category C  
9              2011-11-15        Category C  
10             2011-12-03        Category C  
11             2012-01-05        Category C  
12             2012-01-18        Category C  
13             2012-01-31        Category C  
14             2012-02-06        Category C 

### Task 2

2. Write a query to display the following information for the products, which have not been sold:

- `product_id`
- `product_desc`
- `product_quantity_avail`
- `product_price`
- `Inventory_Value` (product_quantity_avail * product_price)
- `New_Price` after applying a discount as per the criteria below.

Sort the output in decreasing order based on the `Inventory_Value`.

i) If Product Price > 20,000, apply a 20% discount.
ii) If Product Price > 10,000, apply a 15% discount.
iii) If Product Price <= 10,000, apply a 10% discount.

Hint: Use CASE statement, no permanent change in the table required.

[NOTE: TABLES to be used - PRODUCT, ORDER_ITEMS TABLE]


In [18]:
query = """
    SELECT 
        product_id, 
        product_desc, 
        product_quantity_avail, 
        product_price,
        product_quantity_avail * product_price as Inventory_Value,
        CASE
            WHEN product_price > 20000 THEN product_price - (0.20 * product_price)
            WHEN product_price > 10000 THEN product_price - (0.15 * product_price)
            WHEN product_price <= 10000 THEN product_price - (0.10 * product_price)
        END AS New_Price
    FROM orders.product
    JOIN orders.order_items
    USING (product_id)
    ORDER BY product_id
"""

mycursor.execute(query)

# Fetch the results
results = mycursor.fetchall()

# Create a DataFrame from the query results
df = pd.DataFrame(
    results,
    columns=[
        "product_id",
        "product_desc",
        "product_quantity_avail",
        "product_price",
        "Inventory_Value",
        "New_Price",
    ],
)

In [19]:
df

product_id            product_desc  product_quantity_avail product_price  \
0           201       Sky LED 102 CM TV                      30      35000.00   
1           201       Sky LED 102 CM TV                      30      35000.00   
2           201       Sky LED 102 CM TV                      30      35000.00   
3           201       Sky LED 102 CM TV                      30      35000.00   
4           201       Sky LED 102 CM TV                      30      35000.00   
..          ...                     ...                     ...           ...   
200         246           Exam Warriors                      50        100.00   
201         246           Exam Warriors                      50        100.00   
202         246           Exam Warriors                      50        100.00   
203         247      Small Is Beautiful                      40        140.00   
204         248  To Kill a Mocking Bird                      35        210.00   

    Inventory_Value   New_Price  
0        1050000.00  28000.0000  
1        1050000.00  28000.0000  
2        1050000.00  28000.0000  
3        1050000.00  28000.0000  
4        1050000.00  28000.0000  
..              ...         ...  
200         5000.00     90.0000  
201         5000.00     90.0000  
202         5000.00     90.0000  
203         5600.00    126.0000  
204         7350.00    189.0000  

[205 rows x 6 columns]

### Task 3

3. Write a query to display the Product_class_code, Product_class_description, Count of Product type in each product class, and the Inventory Value (product_quantity_avail * product_price). 
The information should be displayed only for those product_class_code that have an Inventory Value of more than 100,000. Sort the output in decreasing order based on the Inventory Value.

[NOTE: TABLES to be used - PRODUCT_CLASS, PRODUCT_CLASS_CODE]


In [20]:
query = """
    SELECT 
        P.PRODUCT_CLASS_CODE,
        P.PRODUCT_DESC,
        PC.PRODUCT_CLASS_DESC AS PRODUCT_TYPE,
        (P.PRODUCT_PRICE * P.PRODUCT_QUANTITY_AVAIL) AS INVENTORY_VALUE
    FROM orders.product P
    JOIN orders.product_class PC
    ON P.PRODUCT_CLASS_CODE = PC.PRODUCT_CLASS_CODE
    WHERE (P.PRODUCT_PRICE * P.PRODUCT_QUANTITY_AVAIL) > 100000
    ORDER BY INVENTORY_VALUE DESC
"""

mycursor.execute(query)

# Fetch the results
results = mycursor.fetchall()

# Create a DataFrame from the query results
df = pd.DataFrame(
    results,
    columns=["PRODUCT_CLASS_CODE", "PRODUCT_DESC", "PRODUCT_TYPE", "INVENTORY_VALUE"],
)

In [21]:
df

PRODUCT_CLASS_CODE                                       PRODUCT_DESC  \
0                 2050                                  Sky LED 102 CM TV   
1                 3000                         Samsung Galaxy Tab 2 P3100   
2                 3000                        Sony Xperia U (Black White)   
3                 2055                              OnePlus 6 Smart Phone   
4                 3000                          Nikon Coolpix L810 Bridge   
5                 3001                        LG MS-2049UW Solo Microwave   
6                 2050               Sams 192 L4 Single-door Refrigerator   
7                 3001                          Nokia Asha 200 (Graphite)   
8                 3001         HP Deskjet 2050 All-in-One - J510a Printer   
9                 2052                         Blossoms Lehenga Choli set   
10                2055                                 Samsung Galaxy On6   
11                3002                   Dell Targus Synergy 2.0 Backpack   
12                3002  Tom Clancy's Ghost Recon: Future Soldier (PC G...   
13                3002                                 Nokia 1280 (Black)   
14                2050                    Jocky Speaker Music System HT32   
15                2057                               Titan Karishma Watch   

              PRODUCT_TYPE INVENTORY_VALUE  
0              Electronics      1050000.00  
1     Promotion-High Value       965000.00  
2     Promotion-High Value       824950.00  
3                  Mobiles       812500.00  
4     Promotion-High Value       749350.00  
5   Promotion-Medium Value       480000.00  
6              Electronics       420000.00  
7   Promotion-Medium Value       407000.00  
8   Promotion-Medium Value       374900.00  
9                  Clothes       300000.00  
10                 Mobiles       280000.00  
11     Promotion-Low Value       249750.00  
12     Promotion-Low Value       249750.00  
13     Promotion-Low Value       249750.00  
14             Electronics       169100.00  
15                 Watches       122395.00

### Task 4 

4.	Write a query to display customer_id, full name, customer_email, customer_phone and country of customers who have cancelled all the orders placed by them (USE SUB-QUERY) \
\
[NOTE: TABLES to be used - ONLINE_CUSTOMER, ADDRESSS, ORDER_HEADER]

In [23]:
query = """
    SELECT
        CUSTOMER_ID,
        CONCAT(CUSTOMER_FNAME, ' ', CUSTOMER_LNAME) AS Full_Name,
        CUSTOMER_EMAIL,
        CUSTOMER_PHONE,
        COUNTRY
    FROM orders.online_customer
    JOIN orders.order_header USING (CUSTOMER_ID)
    JOIN orders.address USING (ADDRESS_ID)
    WHERE ORDER_STATUS = 'Cancelled'
"""

mycursor.execute(query)

# Fetch the results
results = mycursor.fetchall()

# Create a DataFrame from the query results
df = pd.DataFrame(
    results,
    columns=["CUSTOMER_ID", "Full_Name", "CUSTOMER_EMAIL", "CUSTOMER_PHONE", "COUNTRY"],
)
df

CUSTOMER_ID               Full_Name           CUSTOMER_EMAIL  \
0            8        Neetha Castelina       neetha20@gmail.com   
1           33          Niseema Zimmer       niseemaz@yahoo.com   
2           51      Ahmad Bin Gh Azali  ahmad_bingh@yahoo.co.my   
3           34             Hans Zimmer    hans_zimmer@gmail.com   
4           41  Tharman Shanmugaratnam     tharshan@yahoo.co.sg   

   CUSTOMER_PHONE    COUNTRY  
0      8196236362      India  
1      8179413840        USA  
2      7348292313   Malaysia  
3      9477272235        USA  
4      8572898929  Singapore

### Task 5

6.	Write a query to display product_id, product_desc, product_quantity_avail, quantity sold and show inventory Status of products as below as per below condition: 
a. For Electronics and Computer categories, if sales till date is Zero then show 'No Sales in past, give discount to reduce inventory', 
if inventory quantity is less than 10% of quantity sold,show 'Low inventory, need to add inventory', if inventory quantity is less 
than 50% of quantity sold, show 'Medium inventory, need to add some inventory', if inventory quantity is more or equal to 50% of quantity sold, show 'Sufficient inventory' 
b. For Mobiles and Watches categories, if sales till date is Zero then show 'No Sales in past, give discount to reduce inventory', if inventory quantity is less than 20% 
of quantity sold, show 'Low inventory, need to add inventory', if inventory quantity is less than 60% of quantity sold, show 'Medium inventory, need to add some inventory',
 if inventory quantity is more or equal to 60% of quantity sold, show 'Sufficient inventory' c. Rest of the categories, if sales till date is Zero then show 'No Sales in past,
 give discount to reduce inventory', if inventory quantity is less than 30% of quantity sold, show 'Low inventory, need to add inventory', if inventory quantity is less than 
 70% of quantity sold, show 'Medium inventory, need to add some inventory', if inventory quantity is more or equal to 70% of quantity sold, show 'Sufficient inventory' 
 \
 -- (USE SUB-QUERY) -- \
 \
 \
 [NOTE: TABLES to be used - PRODUCT, PRODUCT_CLASS, ORDER_ITEMS]

In [25]:
query = """
    SELECT
        orders.product.PRODUCT_ID,
        orders.product_class.PRODUCT_CLASS_DESC,
        orders.product.PRODUCT_DESC,
        orders.product_class.PRODUCT_CLASS_CODE,
        PRODUCT_QUANTITY_AVAIL,
        PRODUCT_QUANTITY AS QUANTITY_SOLD,
        CASE
            WHEN PRODUCT_CLASS_DESC IN ('Electronics', 'Computer') THEN
                CASE
                    WHEN PRODUCT_QUANTITY_AVAIL < 10 THEN 'Low inventory, need to add inventory'
                    WHEN PRODUCT_QUANTITY_AVAIL <= 50 THEN 'Medium inventory, need to add some inventory'
                    ELSE 'Sufficient inventory'
                END
            WHEN PRODUCT_CLASS_DESC IN ('Mobiles', 'Watches') THEN
                CASE
                    WHEN PRODUCT_QUANTITY_AVAIL < 20 THEN 'Low inventory, need to add inventory'
                    WHEN PRODUCT_QUANTITY_AVAIL <= 60 THEN 'Medium inventory, need to add some inventory'
                    ELSE 'Sufficient inventory'
                END
            WHEN PRODUCT_QUANTITY_AVAIL < 30 THEN 'Low inventory, need to add inventory'
            WHEN PRODUCT_QUANTITY_AVAIL <= 70 THEN 'Medium inventory, need to add some inventory'
            ELSE 'Sufficient inventory'
        END AS INVENTORY_STATS
    FROM orders.product
    JOIN orders.product_class ON orders.product.PRODUCT_CLASS_CODE = orders.product_class.PRODUCT_CLASS_CODE
    JOIN orders.order_items ON orders.product.PRODUCT_ID = orders.order_items.PRODUCT_ID
"""

mycursor.execute(query)

# Fetch the results
results = mycursor.fetchall()

# Create a DataFrame from the query results
df = pd.DataFrame(
    results,
    columns=[
        "PRODUCT_ID",
        "PRODUCT_CLASS_DESC",
        "PRODUCT_DESC",
        "PRODUCT_CLASS_CODE",
        "PRODUCT_QUANTITY_AVAIL",
        "QUANTITY_SOLD",
        "INVENTORY_STATS",
    ],
)

df

PRODUCT_ID PRODUCT_CLASS_DESC                         PRODUCT_DESC  \
0           205            Clothes                Infant Sleepwear Blue   
1           201        Electronics                    Sky LED 102 CM TV   
2           212            Mobiles                   Samsung Galaxy On6   
3           244         Furnitures               Foldable Premium Chair   
4           238         Stationery     Kasyo DJ-2100 Desktop Calculator   
..          ...                ...                                  ...   
200         213              Books                            Alchemist   
201         235         Stationery  Cindy HMPOC Pencil Box (Multicolor)   
202         220               Bags                Hello Kitti Lunch Bag   
203         238         Stationery     Kasyo DJ-2100 Desktop Calculator   
204         240         Stationery                   4M Post It Pad 3.5   

     PRODUCT_CLASS_CODE  PRODUCT_QUANTITY_AVAIL  QUANTITY_SOLD  \
0                  2052                      50              3   
1                  2050                      30              1   
2                  2055                      20              1   
3                  2058                       6              1   
4                  2056                      10              4   
..                  ...                     ...            ...   
200                2054                      50              1   
201                2056                      10              3   
202                2059                      15              1   
203                2056                      10              1   
204                2056                       8              2   

                                  INVENTORY_STATS  
0    Medium inventory, need to add some inventory  
1    Medium inventory, need to add some inventory  
2    Medium inventory, need to add some inventory  
3            Low inventory, need to add inventory  
4            Low inventory, need to add inventory  
..                                            ...  
200  Medium inventory, need to add some inventory  
201          Low inventory, need to add inventory  
202          Low inventory, need to add inventory  
203          Low inventory, need to add inventory  
204          Low inventory, need to add inventory  

[205 rows x 7 columns]

### Task 6

6.	Write a query to display order_id and volume of the biggest order (in terms of volume) that can fit in carton id 10 -- [NOTE: TABLES to be used - CARTON, ORDER_ITEMS, PRODUCT]


In [27]:
query = """
    SELECT 
        *,
        (orders.product.LEN * orders.product.WEIGHT * orders.product.HEIGHT) AS Volume
    FROM orders.order_items
    JOIN orders.product
    USING (PRODUCT_ID)
    WHERE (orders.product.LEN * orders.product.WEIGHT * orders.product.HEIGHT) < 18000000
    ORDER BY (orders.product.LEN * orders.product.WEIGHT * orders.product.HEIGHT) DESC
"""

mycursor.execute(query)

# Fetch the results
results = mycursor.fetchall()

# Create a DataFrame from the query results
df = pd.DataFrame(results, columns=mycursor.column_names)
df

PRODUCT_ID  ORDER_ID  PRODUCT_QUANTITY               PRODUCT_DESC  \
0           201     10001                 1          Sky LED 102 CM TV   
1           201     10009                 1          Sky LED 102 CM TV   
2           201     10012                 1          Sky LED 102 CM TV   
3           201     10014                 1          Sky LED 102 CM TV   
4           201     10020                 1          Sky LED 102 CM TV   
..          ...       ...               ...                        ...   
182         247     10032                 1         Small Is Beautiful   
183         221     10009                 1  Cybershot DWC-W325 Camera   
184         221     10020                 1  Cybershot DWC-W325 Camera   
185         221     10026                 1  Cybershot DWC-W325 Camera   
186         221     10059                 1  Cybershot DWC-W325 Camera   

     PRODUCT_CLASS_CODE PRODUCT_PRICE  PRODUCT_QUANTITY_AVAIL  LEN  WIDTH  \
0                  2050      35000.00                      30  905    750   
1                  2050      35000.00                      30  905    750   
2                  2050      35000.00                      30  905    750   
3                  2050      35000.00                      30  905    750   
4                  2050      35000.00                      30  905    750   
..                  ...           ...                     ...  ...    ...   
182                2054        140.00                      40  180    100   
183                2050       5300.00                       5  100     55   
184                2050       5300.00                       5  100     55   
185                2050       5300.00                       5  100     55   
186                2050       5300.00                       5  100     55   

     HEIGHT   WEIGHT        Volume  
0       700  15.0000  9502500.0000  
1       700  15.0000  9502500.0000  
2       700  15.0000  9502500.0000  
3       700  15.0000  9502500.0000  
4       700  15.0000  9502500.0000  
..      ...      ...           ...  
182      15   0.1000      270.0000  
183      40   0.0500      200.0000  
184      40   0.0500      200.0000  
185      40   0.0500      200.0000  
186      40   0.0500      200.0000  

[187 rows x 12 columns]

### Task 7

7.	Write a query to display product_id, product_desc and total quantity of products which are sold together with product id 201 and are not shipped to city Bangalore and New Delhi. 
Display the output in descending order with respect to the tot_qty. \

-- (USE SUB-QUERY) -- \
\
[NOTE: TABLES to be used - order_items, product,order_head, online_customer, address]


In [30]:
query = """
    SELECT P.PRODUCT_ID,
        PC.PRODUCT_CLASS_CODE AS Product_Class_Code,
        PC.PRODUCT_CLASS_DESC AS Product_Class_Description,
        SUM(OI.PRODUCT_QUANTITY) AS Total_Quantity,
        SUM(OI.PRODUCT_QUANTITY * P.PRODUCT_PRICE) AS Total_Value,
        A.CITY
    FROM orders.order_items OI
    JOIN orders.order_header OH ON OH.ORDER_ID = OI.ORDER_ID 
    JOIN orders.online_customer OC ON OC.CUSTOMER_ID = OH.CUSTOMER_ID
    JOIN orders.product P ON P.PRODUCT_ID = OI.PRODUCT_ID
    JOIN orders.product_class PC ON PC.PRODUCT_CLASS_CODE = P.PRODUCT_CLASS_CODE
    JOIN orders.address A ON A.ADDRESS_ID = OC.ADDRESS_ID 
    WHERE OH.ORDER_STATUS ='Shipped' AND A.CITY NOT IN ('Bangalore','New Delhi') 
    GROUP BY P.PRODUCT_ID, PC.PRODUCT_CLASS_CODE, PC.PRODUCT_CLASS_DESC, A.CITY
    ORDER BY Total_Quantity DESC;
"""

mycursor.execute(query)

# Fetch the results
results = mycursor.fetchall()

# Create a DataFrame from the query results
df = pd.DataFrame(
    results,
    columns=[
        "PRODUCT_ID",
        "Product_Class_Code",
        "Product_Class_Description",
        "Total_Quantity",
        "Total_Value",
        "CITY",
    ],
)
df

PRODUCT_ID  Product_Class_Code Product_Class_Description Total_Quantity  \
0          218                2056                Stationery             15   
1          235                2056                Stationery             15   
2          236                2056                Stationery             10   
3          218                2056                Stationery             10   
4          240                2056                Stationery              7   
..         ...                 ...                       ...            ...   
94         242                2058                Furnitures              1   
95         204                2051                      Toys              1   
96         212                2055                   Mobiles              1   
97         238                2056                Stationery              1   
98         238                2056                Stationery              1   

   Total_Value              CITY  
0       375.00         Hyderabad  
1      1200.00         Hyderabad  
2      3500.00            Mysore  
3       250.00             Salem  
4       245.00             Hosur  
..         ...               ...  
94     3500.00  Kuala Terengganu  
95     4500.00          Brooklyn  
96    14000.00              Avon  
97      338.00          Brooklyn  
98      338.00             Hosur  

[99 rows x 6 columns]

### Task 8 

Write a query to display the order_id,customer_id and customer fullname, total quantity of products shipped for order ids which are even 
and shipped to address where pincode is not starting with "5" -- \
[NOTE: TABLES to be used online_customer,Order_header, order_items,address]

In [31]:
query = """
    SELECT
        OH.ORDER_ID,
        OC.CUSTOMER_ID,
        A.PINCODE,
        CONCAT(OC.CUSTOMER_FNAME, ' ', OC.CUSTOMER_LNAME) AS Customer_FullName,
        OI.PRODUCT_QUANTITY
    FROM
        orders.online_customer OC
        JOIN orders.address A ON OC.ADDRESS_ID = A.ADDRESS_ID
        JOIN orders.order_header OH ON OC.CUSTOMER_ID = OH.CUSTOMER_ID
        JOIN orders.order_items OI ON OH.ORDER_ID = OI.ORDER_ID
    WHERE
        OH.ORDER_ID % 2 = 0
        AND LEFT(A.PINCODE, 1) = '5'
    ORDER BY
        OI.PRODUCT_QUANTITY DESC
"""

mycursor.execute(query)

# Fetch the results
results = mycursor.fetchall()

# Create a DataFrame from the query results
df = pd.DataFrame(
    results,
    columns=[
        "ORDER_ID",
        "CUSTOMER_ID",
        "PINCODE",
        "Customer_FullName",
        "PRODUCT_QUANTITY",
    ],
)
df

ORDER_ID  CUSTOMER_ID  PINCODE  Customer_FullName  PRODUCT_QUANTITY
0      10020           18   517252        Suresh Babu                20
1      10018           11   517247          Vikas Jha                15
2      10018           11   517247          Vikas Jha                15
3      10012            2   560172      Jackson Davis                10
4      10018           11   517247          Vikas Jha                 6
..       ...          ...      ...                ...               ...
67     10012            2   560172      Jackson Davis                 1
68     10014            9   562109      Devika Satish                 1
69     10060           50    50480  Sri binti  Yaakob                 1
70     10014            9   562109      Devika Satish                 1
71     10060           50    50480  Sri binti  Yaakob                 1

[72 rows x 5 columns]